### ISSUE TITLE
Flux unquotes resource content

### ISSUE THREAD

**Issue by MartinEmrich** [fluxcd/flux2#1799]
### Describe the bug

Something in flux v2 seems to "unquote" resource content. Thus the content of e.g. a configmap differes between manual apply and application via Git/Flux.

Note that I make use of a "postBuild:" block with "substitute:", so whatever implements that is part of the processing chain.

### Steps to reproduce

Example configmap:

`````
apiVersion: v1
kind: ConfigMap
metadata:
  name: minimal-configmap
  namespace: eseia
data:
  some.json: |-
    {
      "someregex": "^/system\\.slice/(.+)\\.service$"
    }
`````
note the escaped backslash.

Applying it manually and the reviewing it yields the same content as expected (kubectl apply ...; kubectl get configmap -oyaml ...).

But committing it as a file and having Flux apply it via a GitRepository/Kustomization pair yields this:

`````
data:
  some.json: |-
    {
      "someregex": "^/system\.slice/(.+)\.service$"
    }
`````
Note the single backslashes.

### Expected behavior

I would expect flux to keep the resource contents as-is.

### OS / Distro

K8s 1.20

### Flux version

0.17.0

### Flux check

► checking prerequisites
✔ kubectl 1.22.1 >=1.18.0-0
✔ Kubernetes 1.20.7-eks-d88609 >=1.16.0-0
► checking controllers
✔ helm-controller: deployment ready
► ghcr.io/fluxcd/helm-controller:v0.11.2
✔ image-automation-controller: deployment ready
► ghcr.io/fluxcd/image-automation-controller:v0.14.1
✔ image-reflector-controller: deployment ready
► 364445721381.dkr.ecr.eu-central-1.amazonaws.com/operations/image-reflector-controller:20210624184344
✔ kustomize-controller: deployment ready
► ghcr.io/fluxcd/kustomize-controller:v0.14.0
✔ notification-controller: deployment ready
► ghcr.io/fluxcd/notification-controller:v0.16.0
✔ source-controller: deployment ready
► ghcr.io/fluxcd/source-controller:v0.15.4
✔ all checks passed

### Git provider

N/A

### Container Registry provider

N/A

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by stefanprodan** [comment]
Can you please disable substitution for that configmap and check if the content stays the same?

**Comment by MartinEmrich** [comment]
hello @stefanprodan!

I have no idea how to disable it for a specific configmap or resource, but removing the whole "postBuild:" block from the kustomization fixes it (of course all templated files are now bogus).

Is there a way (annotation, label, ...) to disable it only for one specific file? that would be a viable workaround for now....

Thx
Martin

**Comment by stefanprodan** [comment]
> Is there a way (annotation, label, ...) to disable it only for one specific file?

Yes please see the docs: https://fluxcd.io/docs/components/kustomize/kustomization/#variable-substitution

**Comment by MartinEmrich** [comment]
Thanks, that did work, too, even after reenabling the postBuild: block.

**Comment by stefanprodan** [comment]
Going to close this as wontfix. To fix this bug we would need to fork https://github.com/drone/envsubst someday maybe...

**Comment by mochja** [comment]
@stefanprodan the issue seems to fixed in the upstream:
- https://github.com/drone/envsubst/pull/27
- https://github.com/grafana/agent/pull/1004

**Comment by crinjes** [comment]
This appears to have been fixed with the 2.3.0 release.

We had to write `\\\\n` to end up with the desired `\\n` in the deployed config in a file that requires variable substitution. With 2.3.0 this is no longer the case and the `\\n` in source remains untouched.
